### ISSUE TITLE
Agile template : Creating new defect has by default  Priority P1, Severity- Sev1, and resolution-Done

### ISSUE THREAD

**Issue by Raunak1203** [openshiftio/openshift.io#3832]
- Create a new workItem of type Defect
- Click on workitem to open quickPreview
- By default Priority p1, Severity - Sev1, and resolution - Done selected

![screenshot from 2018-06-20 11-26-59](https://user-images.githubusercontent.com/34271052/41639770-dc555b88-747c-11e8-8581-6d2f567f1824.png)

**Comment by Raunak1203** [comment]
This response is coming from backend

**Comment by jarifibrahim** [comment]
> By default Priority p1, Severity - Sev1, and resolution - Done selected

We should have `None/To Be Triaged` field value as default

**Comment by michaelkleinhenz** [comment]
Yes, that is a known issue. But before we can fix this on the template, we need a UX consideration. So basically the issue is that for enum fields, the *first* value of a list is always the default and there is no "nil" value.

We could play around with the enum value list in the template, like reversing the values or adding a "n/a" "value", but that all appears to me like a hack. So we left this in for now. Let's have a discussion on how to properly treat "nil" values on enum fields..

**Comment by michaelkleinhenz** [comment]
@rgarg1 I'd say that is not always the case. So for example, a component field should never be empty. We need a general solution for this.

**Comment by Veethika** [comment]
IMO these fields should always hold a default value that's most commonly allocated in case of defects and are unalarming(like P1 and Sev1). The following could be considered to show as default:
Severity: Sev3 - Medium 
Priority: P3 - Medium
Resolution: None
@catrobson @nimishamukherjee WDYT?
@michaelkleinhenz we would still need to discuss "how to properly treat "nil" values on enum fields" to solve the problem with 'Story-points'.

**Comment by rgarg1** [comment]
@Veethika 
Having unalarming values as defaults is an acceptable solution. However, we still have to deal with None in the Resolution field and as you mentioned, story-points.

@michaelkleinhenz An empty field, screaming for attention sounds more acceptable to me than being populated with defaults, which was never updated by the author. But this is just my opinion.

**Comment by Veethika** [comment]
@rgarg leaving a field blank makes more sense when the content has to be unique for that field every single time, pre-populating with some default value that fits the majority of the cases can make life easier for the user.

**Comment by michaelkleinhenz** [comment]
That all feels to me like we need a proper "default value" handling in the backend. Currently, simply the first entry is treated as default. Which will not work with the above suggestions. To me it absolutely makes sense to have a default defined there explicitly. Like @Veethika has suggested. But that is actually a new story. I'll put it into the backlog.

**Comment by kwk** [comment]
> That all feels to me like we need a proper "default value" handling in the backend. Currently, simply the first entry is treated as default. Which will not work with the above suggestions. To me it absolutely makes sense to have a default defined there explicitly. Like @Veethika has suggested. But that is actually a new story. I'll put it into the backlog.

@michaelkleinhenz We have the mechanics in place to have defaults. But what is blocking us on moving values up in an Enum to make them the default?

**Comment by rgarg1** [comment]
@michaelkleinhenz This is a SEV-2 and needs an early fix. As @kwk mentioned, this could be done.

**Comment by kwk** [comment]
I'm working on default values for all kind of fields and I'm almost finished.

**Comment by kwk** [comment]
I've made the changes to the template here in #2247:

https://github.com/fabric8-services/fabric8-wit/pull/2247/files#diff-ef9bca69129f8d8e357ad2ba0192455a.

The resolution is already set to `None` for a while already.

**Comment by kwk** [comment]
Verified fix implemented by https://github.com/fabric8-services/fabric8-wit/pull/2247 in prod-preview.

**Comment by Veethika** [comment]
UX suggestion provided, removing UX labels.

**Comment by joshuawilson** [comment]
@kwk I just looked at production and the `resolution` is set to `Done`.

**Comment by kwk** [comment]
> @kwk I just looked at production and the `resolution` is set to `Done`.

In https://github.com/openshiftio/openshift.io/issues/3832#issuecomment-414303809 I wrote that the resolution has been set to `None`  for a while already. As it turns out this was only true for the Impediment type but not for the Defect WIT.

I'll change that right away.

**Comment by kwk** [comment]
@sanbornsen for the Impediment, we have a `None` resolution and now we have one for a Defect as well. I wonder if we should keep the `None` that was added by the UI. For an Impediment we currently see two `None`s. Only one of them is selectable and the other is not. 

![bildschirmfoto von 2018-10-29 12-16-07](https://user-images.githubusercontent.com/193408/47646579-a1150a00-db74-11e8-8a70-d11de0455014.png)

**Comment by kwk** [comment]
@sanbornsen We now have `None` as the default for all new Defects in prod-preview. The response from the work item creation shows that it is `None` for the `resolution` field. Do you know what problem there is within the UI when you see this screenshot?

![bildschirmfoto von 2018-10-30 16-42-44](https://user-images.githubusercontent.com/193408/47730759-1bbc5300-dc63-11e8-9a80-cfbbb551705e.png)

**Comment by kwk** [comment]
@sanbornsen I'm assigning the issue to you because I think that my part is done.

**Comment by jarifibrahim** [comment]
> @sanbornsen for the Impediment, we have a `None` resolution and now we have one for a Defect as well (should be there once [fabric8-services/fabric8-wit#2334](https://github.com/fabric8-services/fabric8-wit/pull/2334) is rolled out to prod-preview). I wonder if we should keep the `None` that was added by the UI. For an Impediment we currently see two `None`s. Only one of them is selectable and the other is not.
> 
> ![bildschirmfoto von 2018-10-29 12-16-07](https://user-images.githubusercontent.com/193408/47646579-a1150a00-db74-11e8-8a70-d11de0455014.png)

This should be fixed by https://github.com/fabric8-ui/fabric8-planner/pull/2798

**Comment by kwk** [comment]
> @kwk please don't assign issues to anyone but yourself. Unless done so in the context of sprint planning and only for your team.

Why not? It is transparent for when and why it did it. I've talked about this before with @sanbornsen so I thought he knows best. If he doesn't want to work on it, he can assign it back or to someone else.

**Comment by joshuawilson** [comment]
Because being assigned means you are actively working on it and no one else should. If he doesn't check this then it would just stay assigned to him and not get worked. In our planning cycle we assign all the work the individual should be able to do. If they have more bandwidth they can come look for unassigned bugs/issues and pick them up.

**Comment by jarifibrahim** [comment]
This issue has been fixed on prod.
